sdk/azidentity/azidentity.go

```diff
@@ -45,21 +55,30 @@ type AuthenticationFailedError struct {
 	Response      *azcore.Response
 }
 
-func (e *AuthenticationFailedError) Error() string {
+func (e *AuthenticationResponseError) Error() string {
 	msg := e.Message
 	if len(e.Description) > 0 {
 		msg += " " + e.Description
 	}
 	return msg
 }
 
+// AuthenticationFailedError is a struct used to marshal responses when authentication has failed
+type AuthenticationFailedError struct {
+	Err error
```

Comment:
This should be `inner error` and provide an `Unwrap() error` method to properly support `As()` and `Is()`.

Reply:
Looks like there should also be a `msg string` field for the case where there is no error.

Reply:
Ok I added an Unwrap method on AuthenticationFailedError and a msg field which is returned when calling the Error method.